src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,224 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> node = new Node<>(value);
+            node.prev = null;
+            node.next = null;
```

Comment:
if you pass three values to the node constructor at once: prev, item, next, then it will not be necessary to initialize these fields separately each time when creating a node

Reply:
ok, i'll fix it. Sounds really more efficient